pennylane/gradients/jvp.py

```diff
@@ -0,0 +1,303 @@
+            j_ = qml.math.cast_like(j_, tangent)
+            jac_new.append(j_)
+        jac = tuple(jac_new)
+    else:
+        jac = qml.math.convert_like(jac, tangent)
+        jac = qml.math.cast_like(jac, tangent)
+    return jac
+
+
+def compute_jvp_single(tangent, jac):
+    """Convenience function to compute the Jacobian vector product for a given
+    tangent vector and a Jacobian for a single measurement tape.
+
+    Args:
+        tangent (list, tensor_like): tangent vector
+        jac (tensor_like, tuple): Jacobian matrix
+
+    Returns:
+        tensor_like: the Jacobian vector product
+
+    **Examples**
+
+    1. For a single parameter and a single measurement without shape (e.g. expval, var):
+
+    .. code-block:: pycon
+
+        >>> tangent = np.array([1.0])
+        >>> jac = np.array(0.2)
+        >>> qml.gradients.compute_jvp_single(tangent, jac)
+        np.array(0.2)
+
+    2. For a single parameter and a single measurment with shape (e.g. probs):
+
+    .. code-block:: pycon
+
+        >>> tangent = np.array([2.0])
+        >>> jac = np.array([0.3, 0.4])
+        >>> qml.gradients.compute_jvp_single(tangent, jac)
+        np.array([0.6, 0.8])
+
+    3. For multiple parameters (in this case 2 parameters) and a single measurement without shape (e.g. expval, var):
+
+    .. code-block:: pycon
+
+        >>> tangent = np.array([1.0, 2.0])
+        >>> jac = tuple([np.array(0.1), np.array(0.2)])
+        >>> qml.gradients.compute_jvp_single(tangent, jac)
+        np.array(0.5)
+
+    4. For multiple parameters (in this case 2 parameters) and a single measurement with shape (e.g. probs):
+
+    .. code-block:: pycon
+
+        >>> tangent = np.array([1.0, 0.5])
+        >>> jac = tuple([np.array([0.1, 0.3]), np.array([0.2, 0.4])])
+        >>> qml.gradients.compute_jvp_single(tangent, jac)
+        np.array([0.2, 0.5])
+
+    """
+    if jac is None:
+        return None
+
+    tangent = qml.math.stack(tangent)
+    jac = _convert(jac, tangent)
+
+    # Single param
+    if not isinstance(jac, tuple):
+        # No trainable parameters
+        if jac.shape == (0,):
+            res = qml.math.zeros((1, 0))
+            return res
+        tangent = qml.math.reshape(tangent, (1,))
+        # No dimension e.g. expval
+        if jac.shape == ():
+            jac = qml.math.reshape(jac, (1,))
+        # With dimension e.g. probs
+        else:
+            jac = qml.math.reshape(jac, (1, -1))
+        res = qml.math.tensordot(jac, tangent, [[0], [0]])
+    # Multiple params
+    else:
+        # No trainable parameters (adjoint)
+        if len(jac) == 0:
+            res = qml.math.zeros((1, 0))
+            return res
+        jac = qml.math.stack(jac)
+        # No dimension e.g. expval
+        if jac[0].shape == ():
+            res = qml.math.tensordot(jac, tangent, 1)
+        # With dimension e.g. probs
+        else:
+            res = qml.math.tensordot(jac, tangent, [[0], [0]])
+    return res
```

Comment:
Very minor: personally, I'd add some new lines here and there, just as this part of the code looks a bit too dense with either a line of code or a comment. This is subjective though.

Reply:
Done!